src/app/core/shared/metadata.utils.ts

```diff
@@ -156,7 +156,7 @@ export class Metadata {
     const outputKeys: string[] = [];
     for (const inputKey of inputKeys) {
       if (inputKey.includes('*')) {
-        const inputKeyRegex = new RegExp('^' + inputKey.replace('.', '\.').replace('*', '.*') + '$');
+        const inputKeyRegex = new RegExp('^' + inputKey.replace('.', '\\.').replace('*', '.*') + '$');
```

Comment:
Doesn't `replace()` just replace the first matching instance?  This RegExp still seems like it might be flawed...as it _seems like_ a RegExp of `'foo.bar.*'` **would incorrectly match** the string `'foo.bard'`.  Could we add a test that proves that this isn't happening, as that's another warning that LGTM told us here: https://lgtm.com/projects/g/DSpace/dspace-angular/snapshot/57a1b26e140ca817fabcd98ab2e094a28f0f9cf4/files/src/app/core/shared/metadata.utils.ts?sort=name&dir=ASC&mode=heatmap#xbfaeb40b2e007884:1

Maybe